lib/interfaces/include/BMSDriverGroup.h

```diff
@@ -0,0 +1,332 @@
+ * LTC6811-2 can operate under broadcast and address functionality, but will only perform address-based commands
+ * Irregardless of which model we choose to use, we need will need access to Arduino and SPI repositories.
+ */
+#include <Arduino.h>
+#include <SPI.h>
+#include <string.h>
+#include <stdio.h>
+#include "Configuration.h"
+#include "LTCSPIInterface.h"
+
+class BMSDriverGroup
+{
+public:
+    BMSDriverGroup() = default;
+    BMSDriverGroup(int cs, int ic_num) : chip_select(cs),
+                                         ic_count(ic_num) {};
+
+    /* -------------------- SETUP FUNCTIONS -------------------- */
+
+    /**
+     * INIT: ONLY CALLED ONCE at initialization. LTC6811 only needs to set up the PEC table
+     * @post We should have initialized the PEC by calling generate_PEC_table(), so every other time
+     * we need to find the PEC it will be much faster.
+     */
+    void init();
+
+    /**
+     * PEC:
+     * The Packet Error Code (PEC) is a Error Checker–like CRC for CAN–to make sure that command and data
+     * bytes are communicated to the device correctly.
+     * This code was taken directly from the data sheet, page 76
+     * @post Instantiate the public pec15Table array of size 256. This variable should never be modified afterwards
+     */
+    void generate_PEC_table();
+
+    /**
+     * WAKEUP:
+     * The Wakeup protocol will move the device from the SLEEP to STANDBY / REFUP state
+     * Reference pages 51-52 on the data sheet for specific information.
+     * @pre we don't care; for us, it's good to perform a wakeup to guarantee proper data propogation
+     * @post essentially sets connected pin to LOW for period of time, then to HIGH for period of time
+     */
+    void start_wakeup_protocol();
+
+    /**
+     * Returns the delay time depending on what state we are in: SLEEP, IDLE, or MEASURE
+     * @return int number of milliseconds
+     */
+    int get_delay_time();
+
+    /* -------------------- READING DATA FUNCTIONS -------------------- */
+
+    /**
+     * Reads voltage data from ALL 4 Cell Voltage Registers on the device.
+     * A register, for those that don't know (I didn't), is memory.
+     * @pre in order to actually "read" anything, we need to call wakeup() and send data over SPI
+     * @post store all 126 cell voltages into the cell_voltages container
+     * AND record the max / minimum values and locations of them
+     */
+    void read_voltages();
+
+    /**
+     * Reads GPIO data from ALL 2 Auxillary GPIO Registers on the device.
+     * For each BMS segment, there is 6 board thermistors and 2 humidity sensors.
+     * NOTE: Conversions are different depending on which we are reading.
+     * @pre in order to actually "read" anything, we need to call wakeup() and send data over SPI
+     * @post store all temperature data into the board_temperatures container
+     * AND record the maximum value and locations
+     */
+    void read_GPIOs();
```

Comment:
why not just have read thermistors and read humidity instead of read gpios? also for this please return the humidities and voltages and dont store them as member variables